x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java

```diff
@@ -405,306 +409,254 @@ public void onFailure(Exception e) {
      * The task executes every step (PIT opening, searches, bulk deletes, PIT closing) using the generic thread pool.
      * The same task instance is used for all the steps and makes sure that a closed instance is not executed again.
      */
-    private class PeriodicMaintenanceTask implements Runnable, Releasable {
-
+    private class PeriodicMaintenanceTask implements Runnable {
         private final TimeValue keepAlive;
         private final int batchSize;
 
-        private final AtomicReference<Exception> error = new AtomicReference<>();
-        private final AtomicBoolean closed = new AtomicBoolean();
+        private final ThrottledTaskRunner taskRunner;
         private final AtomicLong deletes = new AtomicLong();
         private final AtomicLong total = new AtomicLong();
 
-        private volatile Map<String, Set<String>> existingSnapshots;
-        private volatile Set<String> existingRepositories;
-        private final AtomicReference<SearchResponse> searchResponse = new AtomicReference<>();
-        private volatile Instant expirationTime;
-        private volatile String pointIntTimeId;
-        private volatile Object[] searchAfter;
-
         PeriodicMaintenanceTask(TimeValue keepAlive, int batchSize) {
             this.keepAlive = keepAlive;
             this.batchSize = batchSize;
+            this.taskRunner = new ThrottledTaskRunner(this.getClass().getCanonicalName(), 2, threadPool.generic());
```

Comment:
NB slightly more concurrent than it was before: we will carry on running search requests looking for docs to expire in parallel with the bulk request that deletes the docs we already found.